src/mailroom.py

```diff
@@ -0,0 +1,93 @@
+"""Automate thankyou notes and generates donor reports."""
+
+import io
+import sys
+from sys import version_info
+
+from builtins import input
+
+donors = {
+    'Matthew Wade': [50.00, 100.00],
+    'Betty Stewart': [100.00, 50.00, 150.00],
+    'Timothy Berg': [10.00],
+    'Paul Benson': [5]}
+
+
+def get_input():
+    """Function gathers all user input and controls program flow."""
+    # Comply inputs between Python 2 and 3. Default to Python 3:
+    user_input = input
+    # If Python 2:
+    if sys.version_info[:2] <= (2, 7):
+        user_input = raw_input
+    ask_report = user_input("Thank you letter (1), report (2), return to the home menu, (3), or quit? (q) ")
+    if ask_report == '1':
+        thankyou_input = user_input("Do you want a list (1) or to enter a donor name (2) ")
+        while thankyou_input == '1':
+            print("\nDonor List:\n")
+            print(show_list(donors))
+            thankyou_input = user_input("Do you want to see a list (1), enter a donor name (2), or return to the main menu? (3) ")
+        if thankyou_input == '2':
+            name_added = user_input("Please enter the donor name: ")
+            list_check(name_added, donors)
+            donor_amount = None
+            while donor_amount is None:
+                try:
+                    donor_amount = float(user_input("How much did they donate? "))
+                except ValueError:
+                    print("There was something wrong with the amount, please enter a number")
+            add_donation(name_added, donor_amount, donors)
+            print(print_email(name_added, donors))
+            get_input()
+        elif thankyou_input == '3':
+            get_input()
+        else:
+            print("Incorrect input. Please try again. ")
+            get_input()
+    elif ask_report == '2':
+        print('{:<20} {:<20} {:<20} {}'.format('Name:', 'Total Donated', 'Number of Donations', 'Average Donation'))
+        for key in donors:
+            print(print_donor_list(key, donors))
+        get_input()
+    elif ask_report == '3':
+        get_input()
+    elif ask_report == 'q' or ask_report == 'Q':
+        sys.exit()
+    else:
+        print("Incorrect input. Please try again. ")
+        get_input()
```

Comment:
This recursive approach will work in a limited capacity, however I don't recommend that you use this approach in the future. The problem is that your recursion will build a larger and larger function call stack as the program more and more.

If you've made let's say 30 commands in Mailroom Madness, then it throws an error all of a sudden, your error traceback will be 30 frames deep. Not only that, but your computer will be allocating memory to maintaining that deep call stack the whole time.

An iterative approach would be better; under the iterative approach, you have a while loop which runs infinitely until you tell it to break. No call stack depth, no problem.